processor/src/main/java/org/mapstruct/ap/internal/model/source/SourceMethod.java

```diff
@@ -183,9 +183,15 @@ public Builder setDefininingType(Type definingType) {
 
         public SourceMethod build() {
 
-            MappingOptions mappingOptions =
-                new MappingOptions( mappings, iterableMapping, mapMapping, beanMapping, valueMappings );
-
+            MappingOptions mappingOptions = null;
+            if ( (mappings != null && !mappings.isEmpty())
+                || iterableMapping != null
+                || mapMapping != null
+                || beanMapping != null
+                || valueMappings != null ) {
```

Comment:
> That's the thing this one no longer picks up inverse configuration from Mappers in uses (even if they are annotated), and I think that we should not break that.

I tried the same.. By adding a new mapper: `UsedMapper` to `CarMapperWithAutoInheritance`: 

```java
@Mapper(
    uses = UsedMapper.class,
    config = AutoInheritedConfig.class
)
public interface CarMapperWithAutoInheritance {
    CarMapperWithAutoInheritance INSTANCE = Mappers.getMapper( CarMapperWithAutoInheritance.class );

    @Mapping(target = "color", source = "colour")
    CarEntity toCarEntity(CarDto carDto);

    @InheritInverseConfiguration(name = "toCarEntity2")
    CarDto toCarDto(CarEntity entity);

    @Mappings({
        @Mapping(target = "color", source = "colour"),
        @Mapping(target = "auditTrail", constant = "fixed")
    })
    CarEntity toCarEntityWithFixedAuditTrail(CarDto carDto);

    @Mapping(target = "color", source = "colour")
    void intoCarEntityOnItsOwn(CarDto carDto, @MappingTarget CarEntity entity);

    @InheritConfiguration(name = "toCarEntity")
    void intoCarEntity(CarDto carDto, @MappingTarget CarEntity entity);
}
```

```java
@Mapper
public interface UsedMapper {

    @Mapping(target = "color", source = "colour")
    CarEntity toCarEntity2(CarDto carDto);

}
```

And guess.. Null pointer.. :smile:  My best guess is that this method does not belong to this mapper. Since mappers are generated one (and this method does not belong to the mapper at hand) I'm asking myself the question whether we should allow this uses in relation to `@InheritInverseConfiguration`